### PR TITLE
chore(spans): Upgrade SQL parser

### PR DESCRIPTION
**Pull request by jjbayer**
We can move off our own fork now that https://github.com/sqlparser-rs/sqlparser-rs/pull/1140 has been merged & released.

#skip-changelog

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -65,7 +65,10 @@ brotli = "3.3.4"
 bytecount = "0.6.0"
 bytes = "1.4.0"
 cadence = "0.29.0"
-chrono = { version = "0.4.31", default-features = false, features = ["std", "serde"] }
+chrono = { version = "0.4.31", default-features = false, features = [
+    "std",
+    "serde",
+] }
```
Comment (jjbayer) [inline Cargo.toml:71]: EvenBetterToml made some formatting changes in this file.

File: Cargo.toml [hunk 2]
```diff
@@ -146,19 +149,19 @@ sha1 = "0.10.5"
 sha2 = "0.10.6"
 similar-asserts = "1.4.2"
 smallvec = { version = "1.11.2", features = ["serde"] }
-sqlparser = { git = "https://github.com/getsentry/sqlparser-rs", rev = "1bfe51c3db3e5b6bda4f52c56860af378d8ec2d3" }
+sqlparser = { version = "0.44.0" }
```
Comment (jjbayer) [inline Cargo.toml]: @Dav1dde when do we declare features in the workspace? I noticed `sqlparser` has no features listed here, but `smallvec` does.

Comment (olksdr) [inline Cargo.toml]: maybe 
```suggestion
sqlparser = "0.44.0"
```

Since there are no features are enabled.

**Review by olksdr — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (Dav1dde) [reply Cargo.toml]: We should basically never have features in the workspace, only reason you could put them there is it's a feature which always needs to be enabled. I haven't cleaned this up yet, I just moved stuff from the workspace crates to the workspace, but left all the stuff which was configured before.

**Review by Dav1dde — APPROVED** [review]

**Pull request merged**